From 39ee5c5d4d3b757843b841181cab97e1c4f02ae0 Mon Sep 17 00:00:00 2001
From: Jiri Benc <jbenc@redhat.com>
Date: Fri, 24 Oct 2014 20:28:11 -0400
Subject: [net] genetlink: add function genl_has_listeners()

Message-id: <ec55bb60244ca9bce66242c06b325e96fa2c8a74.1414182116.git.jbenc@redhat.com>
Patchwork-id: 98454
O-Subject: [RHEL7.1 net PATCH 03/11] genetlink: add function genl_has_listeners()
Bugzilla: 1110384
RH-Acked-by: Flavio Leitner <fbl@redhat.com>
RH-Acked-by: David S. Miller <davem@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110384
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=8159383

commit 0d566379c5e15a2922dc6bb2ee6a4b7f7a3a0786
Author: Nicolas Dichtel <nicolas.dichtel@6wind.com>
Date:   Thu Sep 18 10:31:03 2014 +0200

    genetlink: add function genl_has_listeners()

    This function is the counterpart of the function netlink_has_listeners().

    Signed-off-by: Nicolas Dichtel <nicolas.dichtel@6wind.com>
    Acked-by: Pravin B Shelar <pshelar@nicira.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/net/genetlink.h b/include/net/genetlink.h
index 16ac93e..d0edc94 100644
--- a/include/net/genetlink.h
+++ b/include/net/genetlink.h
@@ -398,4 +398,12 @@ static inline int genl_set_err(struct genl_family *family, struct net *net,
  return netlink_set_err(net->genl_sock, portid, group, code);
 }
 
+static inline int genl_has_listeners(struct genl_family *family,
+         struct sock *sk, unsigned int group)
+{
+ if (WARN_ON_ONCE(group >= family->n_mcgrps))
+  return -EINVAL;
+ group = family->mcgrp_offset + group;
+ return netlink_has_listeners(sk, group);
+}
 #endif /* __NET_GENERIC_NETLINK_H */
-- 
1.7.1